src/main/java/com/iexec/worker/metric/ComputeDurationsService.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023-2023 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.metric;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.math.stat.descriptive.DescriptiveStatistics;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Simple service to store durations of a compute stage (pre, app or post-compute)
+ * and aggregates them (min, max and average).
+ */
+public class ComputeDurationsService {
+    private static final int STATISTICS_WINDOW = 100;
+
+    private final Map<String, Long> durationPerChainTaskId = new LinkedHashMap<>();
+    private final DescriptiveStatistics statistics;
+
+    public ComputeDurationsService(MeterRegistry registry,
+                                   String workerWalletAddress,
+                                   String context) {
+        statistics = new DescriptiveStatistics(STATISTICS_WINDOW);
+
+        final String[] tags = {"wallet", workerWalletAddress, "phase", context};
+        Gauge.builder(context + "_duration_min", statistics::getMin)
```

Comment:
Naming convention question here: do you want the supplying modules to provide all the model or do you intend to enforce a kind of common prefix ?
Would it be interesting to have something like this to enforce a naming convention in prometheus metrics ?
```suggestion
        Gauge.builder("iexec_" + durationLabel + "_duration_min", statistics::getMin)
```

Reply:
You're totally right. Please see [90aab30](https://github.com/iExecBlockchainComputing/iexec-worker/pull/559/commits/90aab30f50bbcb5af2114d79f2b25142d4d8ca7f).